File: pkg/session/session_state.go
package session

import (
	"time"

	"github.com/phrkdll/monomatch/pkg/card"
	"github.com/phrkdll/monomatch/pkg/player"
)

type SessionState struct {
	Id             SessionId            `json:"id"`
	Name           string               `json:"name"`
	CreatedAt      time.Time            `json:"createdAt"`
	TopMostCard    *card.Card           `json:"topMostCard"`
	RemainingCards int                  `json:"remainingCards"`
	Players        []player.PlayerState `json:"players"`
}

File: frontend/types/session-state.ts
export type SessionStateResponse = {
	id: string
	name: string
	createdAt: Date
	topMostCard: Card
	remainingCards: number
	players: Player[]
}

export type Player = {
	playerName: string
	topMostCard: Card
	cardCount: number
}

export type Card = {
	id: string
	symbols: Symbol[]
}

export type Symbol = {
	id: string
	name: string
}
